extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobNameTest.java

```diff
@@ -0,0 +1,122 @@
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, '',   '', 1, partName",
+            "partName, null,   null, 1, partName",
+            "partName, null,   '', 1, partName",
+            "partName, '',   null, 1, partName",
+    }, nullValues = { "null" })
+    void shouldUsePartName_whenBothBlobNameAndFolderNameIsNullOrEmpty(String partName, String blobName, String folderName, int partSize, String expected) {
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName, folderName, 1, folderName/blobName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithBlobNameUsingSlash_whenPartSizeEqualsOne(String partName, String blobName, String folderName, int partSize, String expected) {
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName,  folderName/, 1,  folderName/blobName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithBlobNameUsingSlash_whenPartSizeEqualsOneAndFolderNameEndsWithSlash(String partName, String blobName, String folderName, int partSize, String expected) {
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName, folderName, 2, folderName/partName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithPartNameUsingSlash_whenPartSizeIsGraterThanOne(String partName, String blobName, String folderName, int partSize, String expected) {
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName,  folderName/, 2,  folderName/partName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithPartNameUsingSlash_whenPartSizeIsGraterThanOneAndFolderNameEndsWithSlash(String partName, String blobName, String folderName, int partSize, String expected) {
+
+        DestinationBlobName destinationBlobName = new DestinationBlobName(blobName, folderName);
+        assertThat(destinationBlobName.resolve(partName, partSize)).isEqualTo(expected);
+    }
```

Comment:
a parameterized test with just a set of parameter is a simple test, please convert them accordingly